src/rgw/rgw_rest_s3.cc

```diff
@@ -3607,11 +3607,13 @@ AWSGeneralAbstractor::get_auth_data_v4(const req_state* const s,
    * aws4_auth_needs_complete and aws4_auth_streaming_mode are set to false
    * by default. We don't need to change that. */
   if (is_v4_payload_unsigned(exp_payload_hash) || is_v4_payload_empty(s)) {
-    return std::make_tuple(access_key_id,
-                           client_signature,
-                           std::move(string_to_sign),
-                           sig_factory,
-                           null_completer_factory);
+    return {
```

Comment:
Huh, [in contrast to `std::pair`, `std::tuple` cannot be implicitly initialized](https://stackoverflow.com/a/14961849). Reworking this.